### DISCUSSION TITLE
MET 10 Not Using Precip Climo File

### DISCUSSION THREAD

**Discussion by craigr3** [dtcenter/METplus#2052] [File I/O] [answered]
I have a precipitation climo file which I want to use as a reference in point_stat.  The file was sent to john's email using DOD Safe.  
Point_stat runs to finish with this climo file but I end up with NA's in the climo_mean, stdev, and cdf fields.
 The climo section of the point_stat config file is:
climo_mean = {
    file_name = ["{CLIMOMEAN1}"];
    field         = [{name = "path to file/cfsr_freq_p6_world_mo_2_mean.nc"; level = "A6"; }];
    regrid = {
         method    = NEAREST;
         width        = 1;
         vld_thresh = 0.5;
         shape       = SQUARE;
    }
    time_interp_method = NEAREST;
    day_interval              = 31;
    hour_interval             = NA; - also tried 6
}
climo_stdev = climo_mean;
climo_std_bins = [1];

Attached is the MET output when running this data.
[runpointstat_galwem_TP.log](https://github.com/dtcenter/METplus/files/10768648/runpointstat_galwem_TP.log)

I noticed in the climo netcdf file that we built them using MET 6 tools.  Would that be a factor? I don't have the raw data anymore so need to get these files to work.  Thanks

**Comment by jprestop** [comment]
Hi @craigr3.  Thank you for your changes.  You mentioned that file was sent to John's email using DOD Safe.  @JohnHalleyGotway is in the process of cutting a release today.  He will follow up here as soon as he is able to.

**Reply by JohnHalleyGotway** [reply]
Hi Bob, yes, I did retrieve those files you sent via DoD Safe. I was out of the office over the holiday weekend and haven't had a chance to jump into the details yet. But I have it on my list of things to do tomorrow. Thanks for your patience.

**Comment by JohnHalleyGotway** [comment, marked as answer]
Bob, you indicated the following setting in your post:
```
climo_mean = {
...
field = [{name = "path to file/cfsr_freq_p6_world_mo_2_mean.nc"; level = "A6"; }];
```
I'm not sure if that's a typo or the actual source of the problem. The `name` should be set to the NetCDF variable name that you want to read from that file. Here you've set it to `"path to file/cfsr_freq_p6_world_mo_2_mean.nc"`. But, using `ncdump` I see the following variable names present in that file:
```
ncdump -h cfsr_freq_p6_world_mo_2_mean.nc | grep "float "
	float CFSR_MEANINCHES_6HOUR_00(lat, lon) ;
	float CFSR_MEANINCHES_6HOUR_06(lat, lon) ;
	float CFSR_MEANINCHES_6HOUR_12(lat, lon) ;
	float CFSR_MEANINCHES_6HOUR_18(lat, lon) ;
```
According to the log you sent, you're evaluating a valid time of 06Z:
```
DEBUG 4:    valid time: 20230213_060000
```
So perhaps you should configure it as follows:
```
climo_mean = {
...
field = [{name = "CFSR_MEANINCHES_6HOUR_06"; level = "A6"; }];
```

Please let me know if that's the underlying issue or whether I should keep digging?

**Comment by craigr3** [comment]
John, that was a typo but it did lead me the real issue of not sending the complete climo var name.  I fixed that and can now read this version of the climo data.  Before closing the ticket, i need to check tbat I can open the thresolded version of the climo data.  Thanks

**Comment by craigr3** [comment]
John, thanks that pointed me to t bg e issue.  You can close the ticket.
